Compilation errors on `openbsd` (`1.5.2`)

**Describe the bug**

Compilation fails on openbsd.

Snippet from [our CI](https://github.com/hashicorp/terraform-ls/runs/6114544196?check_suite_focus=true#step:7:49):
```
# github.com/fsnotify/fsnotify
Error: ../../../go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/kqueue.go:23:6: Watcher redeclared in this block
Error: 	/home/runner/go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/fsnotify_unsupported.go:16:6: previous declaration
Error: ../../../go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/kqueue.go:45:6: NewWatcher redeclared in this block
Error: 	/home/runner/go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/fsnotify_unsupported.go:19:30: previous declaration
Error: ../../../go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/kqueue.go:68:19: (*Watcher).Close redeclared in this block
Error: 	/home/runner/go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/fsnotify_unsupported.go:24:6: previous declaration
Error: ../../../go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/kqueue.go:95:19: (*Watcher).Add redeclared in this block
Error: 	/home/runner/go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/fsnotify_unsupported.go:29:6: previous declaration
Error: ../../../go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/kqueue.go:104:19: (*Watcher).Remove redeclared in this block
Error: 	/home/runner/go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.2/fsnotify_unsupported.go:34:6: previous declaration
```

**To Reproduce**

```sh
GOOS=openbsd GOARCH=amd64 go build
```
```
# github.com/fsnotify/fsnotify
./kqueue.go:23:6: Watcher redeclared in this block
	./fsnotify_unsupported.go:16:6: other declaration of Watcher
./kqueue.go:45:6: NewWatcher redeclared in this block
	./fsnotify_unsupported.go:19:6: other declaration of NewWatcher
./kqueue.go:52:3: unknown field 'kq' in struct literal of type Watcher
./kqueue.go:53:3: unknown field 'watches' in struct literal of type Watcher
./kqueue.go:54:3: unknown field 'dirFlags' in struct literal of type Watcher
./kqueue.go:55:3: unknown field 'paths' in struct literal of type Watcher
./kqueue.go:56:3: unknown field 'fileExists' in struct literal of type Watcher
./kqueue.go:68:19: Watcher.Close redeclared in this block
	./fsnotify_unsupported.go:24:19: other declaration of Close
./kqueue.go:95:19: Watcher.Add redeclared in this block
	./fsnotify_unsupported.go:29:19: other declaration of Add
./kqueue.go:104:19: Watcher.Remove redeclared in this block
	./fsnotify_unsupported.go:34:19: other declaration of Remove
./kqueue.go:56:3: too many errors
```

**Expected behavior**

fsnotify to compile without errors on openbsd.

**Which operating system and version are you using?**

Compiling _for_ `openbsd` using `GOOS` - reproducible from macOS 12.3.1 and `ubuntu-latest` in GitHub Actions Linux VM.

**Additional context**

This came up only with the latest version 1.5.2, 1.5.1 compiled on openbsd without issues.

--- a/fsnotify_unsupported.go
+++ b/fsnotify_unsupported.go
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !darwin && !dragonfly && !freebsd && !linux && !netbsd && !solaris && !windows
-// +build !darwin,!dragonfly,!freebsd,!linux,!netbsd,!solaris,!windows
+//go:build !darwin && !dragonfly && !freebsd && !openbsd && !linux && !netbsd && !solaris && !windows
+// +build !darwin,!dragonfly,!freebsd,!openbsd,!linux,!netbsd,!solaris,!windows
 
 package fsnotify
 

